docker-compose.yml

```diff
@@ -60,13 +63,14 @@ services:
       - "3334:3334"
     depends_on:
       - ml_api
-    command: sh -c 'python manage.py migrate && python manage.py collectstatic -v 2 --noinput && python manage.py runserver --nostatic --noreload 0.0.0.0:3334'
+    command: daphne -b 0.0.0.0 -p 3334 config.routing:application
```

Comment:
Is "daphne" the official django dev server now? last time I check it had a lot of problems.

Also why do we have both "command" here and "ENTRYPOINT" in the Dockerfile? Which one wins? Or both will run, but under different circumstances?

Reply:
Daphne is the default ASGI server used by Django channels. In fact, the previous version was actually using Daphne under the hood because it [overrides](https://channels.readthedocs.io/en/latest/installation.html) the `manage.py runserver` command. This change simply runs the "production" version of Daphne by default. You can still launch the dev server by running `docker compose -f docker-compose.yml -f docker-compose-dev.yml up -d`

Regarding the entrypoint, this is the preferred method of running "startup" commands during container initialization (migrate and collectstatic in this case). Docker actually passes the "command" through as an argument to the entrypoint. Then the last line of `entrypoint.sh` executes the command without spawning a new PID. The entrypoint can be disabled by setting it to an empty string (for example, in the tasks container)